pkg/backup/handlers/arango/backup/arango_client_impl.go

```diff
@@ -0,0 +1,241 @@
+//
+// DISCLAIMER
+//
+// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Lars Maier
+//
+
+package backup
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/arangodb/go-driver"
+	backupApi "github.com/arangodb/kube-arangodb/pkg/apis/backup/v1alpha"
+	database "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1alpha"
+	"github.com/arangodb/kube-arangodb/pkg/util/arangod"
+	"github.com/arangodb/kube-arangodb/pkg/util/errors"
+	"github.com/arangodb/kube-arangodb/pkg/util/k8sutil"
+	"k8s.io/client-go/kubernetes"
+)
+
+type arangoClientBackupImpl struct {
+	deployment *database.ArangoDeployment
+	backup     *backupApi.ArangoBackup
+	driver     driver.Client
+	kubecli    kubernetes.Interface
+}
+
+func newArangoClientBackupFactory(handler *handler) ArangoClientFactory {
+	return func(deployment *database.ArangoDeployment, backup *backupApi.ArangoBackup) (ArangoBackupClient, error) {
+		ctx := context.Background()
+		client, err := arangod.CreateArangodDatabaseClient(ctx, handler.kubeClient.CoreV1(), deployment, false)
+		if err != nil {
+			return nil, err
+		}
+
+		return &arangoClientBackupImpl{
+			deployment: deployment,
+			backup:     backup,
+			driver:     client,
+			kubecli:    handler.kubeClient,
+		}, nil
+	}
+}
+
+func (ac *arangoClientBackupImpl) List() (map[driver.BackupID]driver.BackupMeta, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultArangoClientTimeout)
+	defer cancel()
+
+	backups, err := ac.driver.Backup().List(ctx, nil)
+	if err != nil {
```

Comment:
This implementation is missing the temporary error detection. (just for the record)

Reply:
Temporary error detection is done - on the handler level